lib/internal/buffer.js

```diff
@@ -1,13 +1,780 @@
 'use strict';
 
 const binding = process.binding('buffer');
+const { TypeError, RangeError } = require('internal/errors');
 const { setupBufferJS } = binding;
 
 // Remove from the binding so that function is only available as exported here.
 // (That is, for internal use only.)
 delete binding.setupBufferJS;
 
+// Temporary buffers to convert numbers.
+const float32Array = new Float32Array(1);
+const uInt8Float32Array = new Uint8Array(float32Array.buffer);
+const float64Array = new Float64Array(1);
+const uInt8Float64Array = new Uint8Array(float64Array.buffer);
+
+function checkBounds(buf, offset, byteLength) {
+  checkNumberType(offset);
+  if (buf[offset] === undefined || buf[offset + byteLength] === undefined)
+    boundsError(offset, buf.length - (byteLength + 1));
+}
+
+function checkInt(value, min, max, buf, offset, byteLength) {
+  if (value > max || value < min) {
+    throw new RangeError('ERR_OUT_OF_RANGE',
+                         'value', `>= ${min} and <= ${max}`, value);
+  }
+  checkBounds(buf, offset, byteLength);
+}
+
+function checkNumberType(value, type) {
+  if (typeof value !== 'number') {
+    throw new TypeError('ERR_INVALID_ARG_TYPE',
+                        type || 'offset', 'number', value);
+  }
+}
+
+function boundsError(value, length, type) {
+  if (Math.floor(value) !== value) {
+    checkNumberType(value, type);
+    throw new RangeError('ERR_OUT_OF_RANGE',
+                         type || 'offset', 'an integer', value);
+  }
+
+  if (length < 0)
+    throw new RangeError('ERR_BUFFER_OUT_OF_BOUNDS', null, true);
+
+  throw new RangeError('ERR_OUT_OF_RANGE',
+                       type || 'offset',
+                       `>= ${type ? 1 : 0} and <= ${length}`,
+                       value);
+}
+
+// Container to export all read write functions.
+const readWrites = {};
```

Comment:
It's easier for the next person looking at this code if you write out the exports in one place instead of having that scattered over 100s of lines.

Reply:
Addressed.